addon/internal-session.js

```diff
@@ -110,7 +128,7 @@ export default Ember.ObjectProxy.extend(Ember.Evented, {
     if (!Ember.isEmpty(this.authenticator)) {
       Ember.set(data, 'authenticated', Ember.merge({ authenticator: this.authenticator }, data.authenticated || {}));
     }
-    this.store.persist(data);
+    return this._callStoreAsync('persist', data);
   },
```

Comment:
This shouldn't return the promise. I'm not sure what `setUnknownProperty` is supposed to return but I guess nothing at all.